File: scripts/generate_pdf.js
#!/usr/bin/env node
/**
 * Generate PDF from manual-implementacion-api.md
 * Markdown → HTML (TOC + Mermaid rendered) → PDF via Playwright
 */

const fs = require("fs");
const path = require("path");

const MD_PATH = path.join(__dirname, "..", "docs", "manual-implementacion-api.md");
const HTML_PATH = path.join(__dirname, "..", "docs", "manual-implementacion-api.html");
const PDF_PATH = path.join(__dirname, "..", "docs", "manual-implementacion-api.pdf");

function escapeHtml(s) {
  return s.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;");
}

function slugify(text) {
  return text
    .toLowerCase()
    .normalize("NFD").replace(/[\u0300-\u036f]/g, "")
    .replace(/[^a-z0-9]+/g, "-")
    .replace(/(^-|-$)/g, "");
}

function mdToHtml(md) {
  const lines = md.split("\n");
  let html = "";
  const headings = []; // { level, text, id }
  let i = 0;

  while (i < lines.length) {
    const line = lines[i];

    // --- Code block ---
    if (line.startsWith("```")) {
      const lang = line.slice(3).trim();
      const codeLines = [];
      i++;
      while (i < lines.length && !lines[i].startsWith("```")) {
        codeLines.push(lines[i]);
        i++;
      }
      i++;
      const code = escapeHtml(codeLines.join("\n"));
      if (lang === "mermaid") {
        html += `<div class="mermaid">${codeLines.join("\n")}</div>\n`;
      } else {
        html += `<pre><code class="language-${lang}">${code}</code></pre>\n`;
      }
      continue;
    }

    // --- Table ---
    if (line.trim().startsWith("|") && i + 1 < lines.length && /^\s*\|[-\s:|]+\|\s*$/.test(lines[i + 1])) {
      const headerCells = line.trim().replace(/^\|/, "").replace(/\|$/, "").split("|").map(c => c.trim());
      i += 2;
      const bodyRows = [];
      while (i < lines.length && lines[i].trim().startsWith("|")) {
        const cells = lines[i].trim().replace(/^\|/, "").replace(/\|$/, "").split("|").map(c => c.trim());
        bodyRows.push(cells);
        i++;
      }
      const ths = headerCells.map(c => `<th>${escapeHtml(c)}</th>`).join("");
      const trs = bodyRows.map(cells => {
        const tds = cells.map(c => `<td>${escapeHtml(c)}</td>`).join("");
        return `<tr>${tds}</tr>`;
      }).join("\n");
      html += `<table><thead><tr>${ths}</tr></thead><tbody>\n${trs}\n</tbody></table>\n`;
      continue;
    }

    // --- Heading ---
    const hMatch = line.match(/^(#{1,4})\s+(.*)/);
    if (hMatch) {
      const level = hMatch[1].length;
      const text = hMatch[2].replace(/\*\*/g, "").replace(/`/g, "").trim();
      const id = slugify(text);
      headings.push({ level, text, id });
      // Skip "Tabla de Contenidos" heading (TOC is generated automatically)
      if (/tabla de contenidos/i.test(text)) {
        i++;
        // Also skip the manual TOC list that follows
        while (i < lines.length && (lines[i].trim().startsWith("1.") || lines[i].trim().startsWith("2.") || lines[i].trim().startsWith("3.") || lines[i].trim().startsWith("4.") || lines[i].trim().startsWith("5.") || lines[i].trim().startsWith("6.") || lines[i].trim().startsWith("7.") || lines[i].trim().startsWith("8.") || lines[i].trim().startsWith("9.") || lines[i].trim().startsWith("10.") || lines[i].trim().startsWith("11.") || lines[i].trim().startsWith("12.") || lines[i].trim().startsWith("-") || lines[i].trim() === "")) {
          i++;
        }
        continue;
      }
      html += `<h${level} id="${id}">${inlineFormat(hMatch[2])}</h${level}>\n`;
      i++;
      continue;
    }

    // --- Horizontal rule ---
    if (/^---+\s*$/.test(line)) {
      html += `<hr>\n`;
      i++;
      continue;
    }

    // --- Blockquote ---
    if (line.startsWith(">")) {
      const bqLines = [];
      while (i < lines.length && lines[i].startsWith(">")) {
        bqLines.push(lines[i].replace(/^>\s?/, ""));
        i++;
      }
      // Skip the metadata blockquote (Version, Last update, Architecture, Stack)
      const bqText = bqLines.join(" ");
      if (/Versi[oó]n:/.test(bqText) && /Arquitectura:/.test(bqText)) {
        continue;
      }
      html += `<blockquote>${inlineFormat(bqLines.join(" "))}</blockquote>\n`;
      continue;
    }

    // --- List ---
    if (/^- /.test(line)) {
      const items = [];
      while (i < lines.length && /^- /.test(lines[i])) {
        items.push(`<li>${inlineFormat(lines[i].slice(2))}</li>`);
        i++;
      }
      html += `<ul>\n${items.join("\n")}\n</ul>\n`;
      continue;
    }

    // --- Empty line ---
    if (line.trim() === "") {
      i++;
      continue;
    }

    // --- Paragraph ---
    const paraLines = [];
    while (
      i < lines.length &&
      lines[i].trim() !== "" &&
      !lines[i].startsWith("#") &&
      !lines[i].startsWith("```") &&
      !lines[i].startsWith("|") &&
      !lines[i].startsWith(">") &&
      !lines[i].startsWith("-") &&
      !/^---+\s*$/.test(lines[i])
    ) {
      paraLines.push(lines[i]);
      i++;
    }
    if (paraLines.length > 0) {
      html += `<p>${inlineFormat(paraLines.join(" "))}</p>\n`;
    }
  }

  return { html, headings };
}

function inlineFormat(text) {
  let s = text;
  s = s.replace(/\*\*(.+?)\*\*/g, "<strong>$1</strong>");
  s = s.replace(/\*(.+?)\*/g, "<em>$1</em>");
  s = s.replace(/`([^`]+)`/g, "<code>$1</code>");
  s = s.replace(/\[([^\]]+)\]\(([^)]+)\)/g, '<a href="$2">$1</a>');
  return s;
}

function buildToc(headings) {
  // Skip first h1 (title) and "Tabla de Contenidos" (self-reference)
  const tocHeadings = headings.filter((h, idx) => {
    if (h.level === 1 && idx === 0) return false;
    if (/tabla de contenidos/i.test(h.text)) return false;
    return true;
  });

  let toc = '<nav id="toc">\n<ul class="toc-list">\n';
  for (const h of tocHeadings) {
    const indent = h.level - 2; // h2 = 0 indent, h3 = 1, h4 = 2
    const cls = indent <= 0 ? "toc-item" : `toc-item toc-indent-${indent}`;
    toc += `<li class="${cls}"><a href="#${h.id}">${escapeHtml(h.text)}</a></li>\n`;
  }
  toc += "</ul>\n</nav>\n";
  return toc;
}

function buildFullHtml(tocHtml, bodyHtml) {
  return `<!DOCTYPE html>
<html lang="es">
<head>
<meta charset="UTF-8">
<title>FiscalIA — Manual de Implementación API</title>
<style>
@page { size: A4; margin: 2cm 2.2cm; }
* { box-sizing: border-box; }

body {
  font-family: 'Segoe UI', 'Helvetica Neue', Arial, sans-serif;
  font-size: 11pt;
  line-height: 1.65;
  color: #1a1a1a;
  margin: 0; padding: 0;
}

/* ── Cover Page ── */
.cover {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
  text-align: center;
  page-break-after: always;
  padding: 40px;
}
.cover h1 {
  font-size: 28pt;
  color: #0d1b2a;
  border: none;
  margin-bottom: 10px;
  padding-bottom: 0;
}
.cover h2 {
  font-size: 20pt;
  color: #1565c0;
  border: none;
  margin-bottom: 40px;
  padding-bottom: 0;
}
.cover .divider {
  width: 120px;
  height: 3px;
  background: #1565c0;
  margin: 0 auto;
}

/* ── TOC ── */
#toc {
  page-break-after: always;
  padding-top: 20px;
}
#toc h2 {
  font-size: 18pt; color: #0d1b2a;
  border-bottom: 3px solid #1565c0;
  padding-bottom: 8px; margin-bottom: 20px;
}
.toc-list {
  list-style: none; padding: 0; margin: 0;
}
.toc-item {
  margin: 0; padding: 0;
}
.toc-item a {
  display: block;
  padding: 6px 0 6px 0;
  color: #1a1a1a;
  text-decoration: none;
  border-bottom: 1px dotted #ddd;
  font-size: 10.5pt;
  transition: background 0.15s;
}
.toc-item a:hover {
  background: #e3f2fd;
}
/* h2 entries: bold, larger */
.toc-item a {
  font-weight: 600;
  color: #0d1b2a;
}
/* h3 entries: indented, normal weight */
.toc-indent-1 a {
  padding-left: 20px;
  font-weight: 400;
  color: #333;
  font-size: 10pt;
}
/* h4 entries: double indented, smaller */
.toc-indent-2 a {
  padding-left: 40px;
  font-weight: 400;
  color: #555;
  font-size: 9.5pt;
}

/* ── Headings ── */
h1 {
  font-size: 20pt; color: #0d1b2a;
  border-bottom: 3px solid #1565c0;
  padding-bottom: 8px; margin-top: 32px;
}
h2 {
  font-size: 15pt; color: #1565c0;
  border-bottom: 1px solid #ccc;
  padding-bottom: 4px; margin-top: 28px;
}
h3 {
  font-size: 12pt; color: #2e7d32;
  margin-top: 20px;
}
h4 {
  font-size: 11pt; color: #e65100;
  margin-top: 16px;
}

/* ── Tables ── */
table {
  width: 100%; border-collapse: collapse;
  margin: 12px 0; font-size: 9.5pt;
}
th {
  background-color: #0d1b2a; color: #fff;
  padding: 7px 10px; text-align: left; font-weight: 600;
}
td {
  padding: 5px 10px; border-bottom: 1px solid #e0e0e0;
  vertical-align: top; color: #1a1a1a;
}
tr:nth-child(even) { background-color: #f8f9fa; }

/* ── Code ── */
code {
  font-family: 'Cascadia Code', 'Fira Code', Consolas, monospace;
  background: #f0f0f0; padding: 1px 5px; border-radius: 3px;
  font-size: 9.5pt; color: #b71c1c;
}
pre {
  background: #1e1e2e; color: #cdd6f4;
  padding: 14px 16px; border-radius: 6px;
  font-size: 8.5pt; line-height: 1.5;
  margin: 12px 0;
  white-space: pre-wrap;
  word-break: break-all;
  overflow-wrap: break-word;
  page-break-inside: avoid;
}
pre code {
  background: transparent; color: #cdd6f4;
  padding: 0; font-size: 8.5pt;
  white-space: pre-wrap;
  word-break: break-all;
}

/* ── Blockquote ── */
blockquote {
  border-left: 4px solid #1976d2;
  margin: 14px 0; padding: 10px 16px;
  background: #e8f4fd; color: #0d47a1;
  border-radius: 0 6px 6px 0;
  font-style: italic;
}
blockquote strong { color: #0d47a1; }

/* ── Lists ── */
ul, ol { margin: 8px 0; padding-left: 24px; }
li { margin-bottom: 4px; }

/* ── Misc ── */
strong { color: #0d1b2a; }
hr { border: none; border-top: 1px solid #ccc; margin: 20px 0; }
p { margin: 6px 0; }
a { color: #1565c0; text-decoration: none; }

/* ── Mermaid ── */
.mermaid {
  background: #f8fafb; border: 1px solid #cfd8dc;
  border-radius: 8px; padding: 12px;
  margin: 14px 0; text-align: center;
  page-break-inside: auto;
  overflow: visible;
}
.mermaid svg {
  max-width: 100%; height: auto;
  max-height: 500px;
}
</style>
</head>
<body>
<div class="cover">
  <h1>Manual de Implementación</h1>
  <h2>FiscalIA API</h2>
  <div class="divider"></div>
</div>
${tocHtml}
${bodyHtml}
<script type="module">
  import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@11/dist/mermaid.esm.min.mjs';
  mermaid.initialize({
    startOnLoad: true,
    theme: 'base',
    themeVariables: {
      primaryColor: '#e3f2fd',
      primaryTextColor: '#0d1b2a',
      primaryBorderColor: '#1565c0',
      lineColor: '#1565c0',
      fontSize: '13px'
    },
    securityLevel: 'loose',
    flowchart: { useMaxWidth: true, htmlLabels: true, curve: 'cardinal' },
    sequence: { useMaxWidth: true, actorMargin: 50, mirrorActors: false },
    mindmap: { useMaxWidth: true },
    er: { useMaxWidth: true }
  });
</script>
</body>
</html>`;
}

async function main() {
  const md = fs.readFileSync(MD_PATH, "utf-8");
  const { html: bodyHtml, headings } = mdToHtml(md);

  console.log(`Headings encontrados: ${headings.length}`);
  headings.forEach(h => console.log(`  ${"  ".repeat(h.level - 1)}h${h.level} → #${h.id}`));

  const tocHtml = buildToc(headings);
  const fullHtml = buildFullHtml(tocHtml, bodyHtml);

  fs.writeFileSync(HTML_PATH, fullHtml, "utf-8");
  console.log(`\nHTML generado: ${HTML_PATH}`);

  const { chromium } = require("playwright");
  const browser = await chromium.launch();
  const page = await browser.newPage();

  await page.setContent(fullHtml, { waitUntil: "networkidle" });

  // Wait for Mermaid
  console.log("Esperando renderizado de Mermaid...");
  const diagramCount = (fullHtml.match(/class="mermaid"/g) || []).length;
  console.log(`Diagramas Mermaid: ${diagramCount}`);

  try {
    await page.waitForFunction(
      (expected) => document.querySelectorAll(".mermaid svg").length >= expected,
      diagramCount,
      { timeout: 30000 }
    );
    console.log("Todos los diagramas renderizados.");
  } catch {
    const rendered = await page.evaluate(() => document.querySelectorAll(".mermaid svg").length);
    console.log(`Timeout: ${rendered}/${diagramCount} renderizados.`);
  }

  // Generate PDF
  await page.pdf({
    path: PDF_PATH,
    format: "A4",
    margin: { top: "2cm", bottom: "2cm", left: "2.2cm", right: "2.2cm" },
    printBackground: true,
    displayHeaderFooter: true,
    headerTemplate: '<div style="font-size:8px;color:#999;width:100%;text-align:center;margin-top:8px;">FiscalIA — Manual de Implementación API</div>',
    footerTemplate: '<div style="font-size:8px;color:#999;width:100%;text-align:center;margin-bottom:8px;">Página <span class="pageNumber"></span> de <span class="totalPages"></span></div>',
  });

  await browser.close();
  console.log(`\nPDF generado: ${PDF_PATH}`);
  const stats = fs.statSync(PDF_PATH);
  console.log(`Tamaño: ${(stats.size / 1024).toFixed(0)} KB`);
}

main().catch((err) => {
  console.error("Error:", err);
  process.exit(1);
});

File: scripts/generate_pdf.py
#!/usr/bin/env python3
"""Generate PDF from manual-implementacion-api.md using fpdf2."""

import re
from fpdf import FPDF

MD_PATH = "docs/manual-implementacion-api.md"
PDF_PATH = "docs/manual-implementacion-api.pdf"


FONT_DIR = "C:/Windows/Fonts"


class ManualPDF(FPDF):
    def __init__(self):
        super().__init__()
        # Register Unicode fonts
        self.add_font("Arial", "", FONT_DIR + "/arial.ttf", uni=True)
        self.add_font("Arial", "B", FONT_DIR + "/arialbd.ttf", uni=True)
        self.add_font("Arial", "I", FONT_DIR + "/ariali.ttf", uni=True)
        self.add_font("Arial", "BI", FONT_DIR + "/arialbi.ttf", uni=True)
        self.add_font("Mono", "", FONT_DIR + "/consola.ttf", uni=True)

    def header(self):
        if self.page_no() > 1:
            self.set_font("Arial", "I", 8)
            self.set_text_color(100, 100, 100)
            self.cell(0, 8, "FiscalIA — Manual de Implementación API", align="C")
            self.ln(4)
            self.set_draw_color(200, 200, 200)
            self.line(20, 14, self.w - 20, 14)
            self.ln(4)

    def footer(self):
        self.set_y(-15)
        self.set_font("Arial", "I", 8)
        self.set_text_color(100, 100, 100)
        self.cell(0, 10, f"Página {self.page_no()}/{{nb}}", align="C")

    def chapter_title(self, title, level=1):
        colors = {
            1: (13, 27, 42),
            2: (21, 101, 192),
            3: (46, 125, 50),
            4: (230, 81, 0),
        }
        r, g, b = colors.get(level, (0, 0, 0))
        sizes = {1: 16, 2: 13, 3: 11, 4: 10}
        sz = sizes.get(level, 10)

        self.set_font("Arial", "B", sz)
        self.set_text_color(r, g, b)
        self.multi_cell(0, sz * 0.5, title)
        if level <= 2:
            self.set_draw_color(r, g, b)
            self.line(self.get_x(), self.get_y(), self.get_x() + 170, self.get_y())
        self.ln(3)

    def body_text(self, text):
        self.set_font("Arial", "", 9)
        self.set_text_color(30, 30, 30)
        self.multi_cell(0, 5, text)
        self.ln(2)

    def bold_text(self, text):
        self.set_font("Arial", "B", 9)
        self.set_text_color(30, 30, 30)
        self.multi_cell(0, 5, text)
        self.ln(1)

    def code_block(self, code):
        self.set_font("Mono", "", 7.5)
        self.set_fill_color(240, 240, 240)
        self.set_text_color(40, 40, 40)
        x = self.get_x()
        w = self.w - 2 * self.l_margin
        lines = code.strip().split("\n")
        # Estimate height
        line_h = 4
        block_h = len(lines) * line_h + 6
        if self.get_y() + block_h > self.h - 25:
            self.add_page()
        self.set_x(x)
        self.set_fill_color(245, 245, 245)
        self.rect(x, self.get_y(), w, block_h, "F")
        self.ln(3)
        for line in lines[:80]:  # Limit long blocks
            self.set_x(x + 4)
            self.cell(w - 8, line_h, line[:120])
            self.ln(line_h)
        self.ln(3)

    def mermaid_placeholder(self):
        self.set_font("Arial", "I", 8)
        self.set_fill_color(230, 240, 250)
        self.set_text_color(100, 130, 160)
        w = self.w - 2 * self.l_margin
        self.cell(w, 8, "[ Diagrama Mermaid — ver versión interactiva en el .md original ]", fill=True, align="C")
        self.ln(6)

    def table_row(self, cells, is_header=False):
        if is_header:
            self.set_font("Arial", "B", 8)
            self.set_fill_color(13, 27, 42)
            self.set_text_color(255, 255, 255)
        else:
            self.set_font("Arial", "", 8)
            self.set_text_color(30, 30, 30)

        col_w = (self.w - 2 * self.l_margin) / max(len(cells), 1)
        for cell in cells:
            txt = str(cell)[:60]
            self.cell(col_w, 6, txt, border=0, fill=is_header, align="L")
        self.ln(6)
        if is_header:
            self.set_draw_color(200, 200, 200)
            self.line(self.l_margin, self.get_y(), self.w - self.r_margin, self.get_y())
            self.ln(1)


def parse_md(md_path: str) -> list[tuple[str, str]]:
    """Parse markdown into a list of (type, content) tuples."""
    with open(md_path, "r", encoding="utf-8") as f:
        lines = f.readlines()

    elements = []
    i = 0
    while i < len(lines):
        line = lines[i].rstrip("\n")

        # Heading
        m = re.match(r"^(#{1,4})\s+(.*)", line)
        if m:
            level = len(m.group(1))
            elements.append(("heading", (level, m.group(2).strip())))
            i += 1
            continue

        # Code block
        if line.startswith("```"):
            lang = line[3:].strip()
            code_lines = []
            i += 1
            while i < len(lines) and not lines[i].rstrip("\n").startswith("```"):
                code_lines.append(lines[i].rstrip("\n"))
                i += 1
            i += 1  # skip closing ```
            code = "\n".join(code_lines)
            if lang == "mermaid":
                elements.append(("mermaid", code))
            else:
                elements.append(("code", code))
            continue

        # Table
        if line.strip().startswith("|") and i + 1 < len(lines) and re.match(r"^\s*\|[-\s|]+\|\s*$", lines[i + 1]):
            table_lines = []
            while i < len(lines) and lines[i].strip().startswith("|"):
                row = [c.strip() for c in lines[i].strip().strip("|").split("|")]
                table_lines.append(row)
                i += 1
            elements.append(("table", table_lines))
            continue

        # Horizontal rule
        if re.match(r"^---+\s*$", line):
            elements.append(("hr", ""))
            i += 1
            continue

        # Blockquote
        if line.startswith(">"):
            bq_lines = []
            while i < len(lines) and lines[i].startswith(">"):
                bq_lines.append(lines[i].lstrip("> ").rstrip("\n"))
                i += 1
            elements.append(("blockquote", " ".join(bq_lines)))
            continue

        # Regular text
        if line.strip():
            text_lines = [line]
            i += 1
            while i < len(lines) and lines[i].strip() and not lines[i].startswith("#") and not lines[i].startswith("```") and not lines[i].startswith("|") and not lines[i].startswith(">") and not re.match(r"^---+$", lines[i]):
                text_lines.append(lines[i].rstrip("\n"))
                i += 1
            elements.append(("text", " ".join(text_lines)))
        else:
            i += 1

    return elements


def generate_pdf(md_path: str, pdf_path: str):
    elements = parse_md(md_path)

    pdf = ManualPDF()
    pdf.alias_nb_pages()
    pdf.set_auto_page_break(auto=True, margin=20)

    # Title page
    pdf.add_page()
    pdf.ln(40)
    pdf.set_font("Arial", "B", 28)
    pdf.set_text_color(13, 27, 42)
    pdf.cell(0, 15, "FiscalIA", align="C")
    pdf.ln(18)
    pdf.set_font("Arial", "", 16)
    pdf.set_text_color(21, 101, 192)
    pdf.cell(0, 10, "Manual de Implementación API", align="C")
    pdf.ln(15)
    pdf.set_font("Arial", "", 11)
    pdf.set_text_color(100, 100, 100)
    pdf.cell(0, 8, "Versión 2.0.0 — Julio 2026", align="C")
    pdf.ln(6)
    pdf.cell(0, 8, "Arquitectura: Hexagonal (Ports & Adapters) + DDD", align="C")
    pdf.ln(6)
    pdf.cell(0, 8, "Stack: Python 3.12+ / FastAPI / asyncpg / PostgreSQL 16+", align="C")
    pdf.ln(30)
    pdf.set_draw_color(21, 101, 192)
    pdf.line(60, pdf.get_y(), pdf.w - 60, pdf.get_y())
    pdf.ln(10)
    pdf.set_font("Arial", "I", 10)
    pdf.set_text_color(120, 120, 120)
    pdf.cell(0, 8, "Municipio de Valledupar — Departamento del Cesar", align="C")

    for etype, content in elements:
        # Skip title metadata (already on cover)
        if etype == "text" and content.startswith("> **Versión:"):
            continue
        if etype == "text" and content.startswith("> **Última"):
            continue
        if etype == "text" and content.startswith("> **Arquitectura:"):
            continue
        if etype == "text" and content.startswith("> **Stack:"):
            continue
        if etype == "heading" and content[1] == "Manual de Implementación — FiscalIA API":
            continue
        if etype == "hr":
            continue

        if etype == "heading":
            level, title = content
            # Check if we need a new page for level 1
            if level == 1 and pdf.get_y() > 50:
                pdf.add_page()
            elif level <= 2 and pdf.get_y() > pdf.h - 40:
                pdf.add_page()
            pdf.chapter_title(title, level)

        elif etype == "text":
            # Strip markdown bold/italic for plain text
            clean = re.sub(r"\*\*(.*?)\*\*", r"\1", content)
            clean = re.sub(r"\*(.*?)\*", r"\1", clean)
            clean = re.sub(r"`(.*?)`", r"\1", clean)
            pdf.body_text(clean)

        elif etype == "code":
            pdf.code_block(content)

        elif etype == "mermaid":
            pdf.mermaid_placeholder()

        elif etype == "table":
            rows = content
            if len(rows) > 0:
                # Check if page break needed
                est_h = len(rows) * 6 + 10
                if pdf.get_y() + est_h > pdf.h - 25:
                    pdf.add_page()
                for idx, row in enumerate(rows):
                    pdf.table_row(row, is_header=(idx == 0))
                pdf.ln(4)

        elif etype == "blockquote":
            pdf.set_font("Arial", "I", 9)
            pdf.set_text_color(13, 71, 161)
            pdf.set_fill_color(227, 242, 253)
            w = pdf.w - 2 * pdf.l_margin
            x = pdf.get_x()
            pdf.rect(x, pdf.get_y(), w, 8, "F")
            pdf.cell(w, 8, f"  {content[:100]}")
            pdf.ln(10)

    pdf.output(pdf_path)
    print(f"PDF generado: {pdf_path}")


if __name__ == "__main__":
    generate_pdf(MD_PATH, PDF_PATH)
